tests/h/views/accounts_test.py

```diff
@@ -660,6 +660,29 @@ def test_get_sets_subscriptions_data_in_form(self,
             'notifications': set(['reply']),
         })
 
+    def test_get_for_user_without_email_address(self,
+                                                form_validating_to,
+                                                pyramid_request):
+        controller = views.NotificationsController(pyramid_request)
+        controller.form = form_validating_to({})
+
+        result = controller.get()
+
+        assert 'form' not in result
+        assert result['user_has_email_address'] is None
+
+    def test_get_for_user_with_email_address(self,
+                                             form_validating_to,
+                                             pyramid_request):
+        controller = views.NotificationsController(pyramid_request)
+        controller.form = form_validating_to({})
+        pyramid_request.user = mock.Mock(username='janedoe')
+
+        result = controller.get()
+
+        assert 'form' in result
+        assert result['user_has_email_address'] == pyramid_request.user.email
```

Comment:
Because right now the `user` is just a mock, it will always "act" like it has an `email` attribute, even if it doesn't (mocks allow access to any attr name). That's another good reason to use a factory-generated User model (see above).

Reply:
👍